### PR TITLE
Do not quote hyphenated strings unnecessarily.

### LINKED ISSUE #54: Hyphenated strings are quoted unnecessarily
e.g. 

- foo-bar: bar-foo

is output as:

- "foo-bar": "bar-foo"

It looks like the needs_quote function doesn't really match what the spec mandates.

### PR DESCRIPTION
**Pull request by hdevalke**
fixes #54

### CODE REVIEW

**Comment by chyh1990** [comment]
emm...yaml string quoting rules is quite tricky, and we add these quotes for safety.

I can recall that we introduced some regression regarding to quotes in earlier release, so we have to do this carefully.

@hdevalke can you give some reference (e.g. quoting the yaml spec) about this change ?

@hoodie can you help reviewing this PR?

**Comment by hdevalke** [comment]
The yaml spec for plain style is [here](http://yaml.org/spec/1.2/spec.html#id2788859). However, I think more tests should be added. I think it would not quote `: - ()` while it should.

**Comment by hoodie** [comment]
I poked this PR a bit and could not find anything blatantly bad about this. in fact it seems you can have strings starting with `-` too, as long as it's not followed by a space. however it's late at night, so take this with a grain of salt. I would add this test though:

```rust
    #[test]
    fn test_emitter_quotation() {
        let s = r#"---
a-b: ~
c: 1-2-a
d: thisis &notareference
e: "thisis #notaquote"
f: a:colon
g: a> greater than
h: "some(){thing:Looking;like();`code`}#whichisn'tcode"
i: and | what is this? = false
j: and | what is this? = false
k: thisisno * list
l: 2 + 2 = 5
m: true = false
n: yellow + blue => green <= purple
p: "-foobar""#;

        let docs = YamlLoader::load_from_str(&s).unwrap();
        let doc = &docs[0];
        let mut writer = String::new();

        {
            let mut emitter = YamlEmitter::new(&mut writer);
            emitter.dump(doc).unwrap();
        }
        println!("original:\n{}", s);
        println!("emitted:\n{}", writer);

        assert_eq!(s, writer);
    }

}
```

PS: futurenote: thinking about it, we might want to turn a bunch of these parse and reemit tests into a sweet of yaml files. This would allow us to run them against other yaml implementations as well.

File: src/emitter.rs [hunk 1]
```diff
@@ -300,9 +302,15 @@ fn need_quotes(string: &str) -> bool {
 
     string == ""
     || need_quotes_spaces(string)
+    || string.starts_with(|character: char| {
+        match character {
+            ':' | '&' | '*' | '?' | '|' | '-' | '<' | '>' | '=' | '!' | '%' | '@' => true,
```
Comment (hoodie) [inline src/emitter.rs]: as mentioned in my comment: string may technically start with a `-` as long as they aren't followed by a space.

Comment (chyh1990) [reply src/emitter.rs]: i think this is good enough.

**Pull request merged**